Husky/lint staged

NicoZweifel opened the pull request:
- Resolves https://github.com/mintlify/components/issues/65
- Moves headlessui to dev/peerdependencies
- Adds eslint config

We could move the eslint config to a `@mintlify/eslint-config-react` package:

```json
{
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:react-hooks/recommended",
    "@mintlify/eslint-config-typescript"
  ]
}
```

linting all files shows: 
![image](https://user-images.githubusercontent.com/34443492/220155516-f95fef8c-3895-4552-a247-333d335bb1b9.png)

We can fix them when changing the files, I added a section to the readme linking to Contribution guide/tools.

benjamin-shen commented on README.md:
```diff
@@ -48,7 +48,13 @@ This repo contains the components Mintlify uses in our open-source documentation
-`npm install @mintlify/components`
+### npm
+
+`npm install @headlessui/react @mintlify/components`
```
Prefer `install-peerdeps`, see https://github.com/mintlify/config/tree/main/packages/prettier-config as an example

NicoZweifel replied:
Yea that's nice. I'll use that. I'll do it similarly to the `config` example and keep both methods, as some people might be on different react versions or on `preact` and don't need to install all peerdependencies.

benjamin-shen commented on README.md:
```diff
@@ -66,11 +72,27 @@ The project is designed for use with static side rendering where we don't have a
 # Documentation
 
-The [repo wiki](https://github.com/mintlify/components/wiki) explains how to contribute to the repo.
+Go to [mintlify.com/docs/components](https://mintlify.com/docs/components) for how to use the components in Mintlify's documentation platform.
```
```suggestion
Go to [mintlify.com/docs/components](https://mintlify.com/docs/components) to see how to use the components in Mintlify's documentation platform.
```

benjamin-shen commented on package.json:
```diff
@@ -18,15 +18,17 @@
     "format": "prettier \"**/*.{css,js,ts,tsx,md}\" --write",
     "format:check": "prettier \"**/*.{css,js,ts,tsx,md}\" --check",
-    "prepare": "npm run build",
+    "prepare": "husky install && npm run build",
```
Can you make this change while you're at it?
```suggestion
    "prepare": "husky install && yarn build",
```

benjamin-shen commented on package.json:
```diff
@@ -18,15 +18,17 @@
     "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook"
+    "build-storybook": "build-storybook",
+    "lint": "eslint src"
```
Can you add `--cache`?

NicoZweifel replied:
done

benjamin-shen reviewed (commented):
Thanks! I like the documentation updates 😄 

I'll make the shared eslint config for react.

hahnbeelee commented on src/Code/CodeBlock.tsx:
```diff
@@ -1,5 +1,5 @@
 import clsx from 'clsx';
-import { ComponentPropsWithoutRef, ForwardedRef, forwardRef, ReactElement } from 'react';
+import React, { ComponentPropsWithoutRef, ForwardedRef, forwardRef, ReactElement } from 'react';
```
Why did you import React here?

NicoZweifel replied:
It is to fix this eslint error:  ` 40:5   error  'React' must be in scope when using JSX  react/react-in-jsx-scope`.
The `React` import is not needed in next.js apps but in other react apps it is sometimes required for it to work.

hahnbeelee replied:
Oo gotcha makes sense

hahnbeelee commented on .eslintrc.json:
```diff
@@ -4,5 +4,8 @@
-  ]
+  ],
+  "rules": {
+    "react/prop-types": "off"
```
What is this for?

NicoZweifel replied:
It disables a rule that checks for prop types, but since we are using typescript we don't use proptypes and don't need the rule.

NicoZweifel commented:
I merged `main`, disabled a rule, fixed some eslint warnings on staged files and updated the docs.

I think this is now good to go. Thanks for your input 👍
